docs/index.bs

```diff
@@ -2970,16 +2972,34 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
               1. For each |eventType| of |settingsObject|'s [=environment settings object/global object=]'s associated list of <a>event listeners</a>' event types:
                   1. [=set/Append=] |eventType| to |workerGlobalScope|'s associated [=ServiceWorkerGlobalScope/service worker=]'s <a>set of event types to handle</a>.
 
-                  Note: If the global object's associated list of event listeners does not have any event listener added at this moment, the service worker's set of event types to handle remains an empty set. The user agents are encouraged to show a warning that the event listeners must be added on the very first evaluation of the worker script.
+                  Note: If the global object's associated list of event listeners does not have any event listener added at this moment, the service worker's set of event types to handle remains an empty set.
 
               1. Set |script|'s <a>has ever been evaluated flag</a>.
+              1. If the [=No-Op Handler identification=] algorithm with |serviceWorker| returns true, the [=has had non-no-op fetch listeners flag=] is set to false.
           1. Run the <a>responsible event loop</a> specified by |settingsObject| until it is destroyed.
           1. Empty |workerGlobalScope|'s <a>list of active timers</a>.
       1. Wait for |serviceWorker| to be [=running=], or for |startFailed| to be true.
       1. If |startFailed| is true, then return *failure*.
       1. Return |serviceWorker|'s [=start status=].
   </section>
 
+  <section algorithm>
+    <h3 id="no-op-handler-identification-algorithm"><dfn>No-Op Handler identification</dfn></h3>
+
+      : Input
+      :: |serviceWorker|, a [=/service worker=]
+      : Output
+      :: a boolean
+
+      1. If |service worker|'s set of event types to handle does not contain "fetch", then returns false.
+      1. If the callback's function bodies (https://tc39.es/ecma262/#prod-FunctionBody) of all fetch event listener in |service worker| are empty, then the user agent *may* return true.
```

Comment:
Things like "callback" need to be referenced properly. Eg, see https://dom.spec.whatwg.org/#event-listener-callback - you'll need to handle cases where the 'callback' is actually an `EventListener` object.

`|service worker|` should be `|serviceWorker|` (as defined in the input)

Rather than just paste the link of https://tc39.es/ecma262/#prod-FunctionBody, create a proper reference. See the `<pre class="anchors">` section at the top of the document.

"of all fetch event listener" needs to be defined correctly in terms of https://dom.spec.whatwg.org/#concept-event-listener.

"empty" needs to be defined more robustly.

Reply:
I am not so confident to write references, but I tried to address this.  please take a look.

Reply:
"event listeners" seems to be already defined.  Do you want to replace all of them with DOM standard's event listener?

Reply:
To avoid the conflict of the "event listeners" name, I avoided to use it while referring callback.

Reply:
Please feel free to reopen if this still matters.